src/main/java/de/tum/in/www1/artemis/service/GradingScaleService.java

```diff
@@ -0,0 +1,123 @@
+package de.tum.in.www1.artemis.service;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.GradeStep;
+import de.tum.in.www1.artemis.domain.GradingScale;
+import de.tum.in.www1.artemis.repository.GradeStepRepository;
+import de.tum.in.www1.artemis.repository.GradingScaleRepository;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+@Service
+public class GradingScaleService {
+
+    private final GradingScaleRepository gradingScaleRepository;
+
+    private final GradeStepRepository gradeStepRepository;
+
+    public GradingScaleService(GradingScaleRepository gradingScaleRepository, GradeStepRepository gradeStepRepository) {
+        this.gradeStepRepository = gradeStepRepository;
+        this.gradingScaleRepository = gradingScaleRepository;
+    }
+
+    /**
+     * Maps a grade percentage to a valid grade step within the grading scale or throws an exception if no match was found
+     *
+     * @param percentage the grade percentage to be mapped
+     * @param gradingScaleId the identifier for the grading scale
+     * @return grade step corresponding to the given percentage
+     */
+    public GradeStep matchPercentageToGradeStep(double percentage, Long gradingScaleId) {
+        if (percentage < 0 || percentage > 100) {
```

Comment:
Why is it invalid if it is == 0 or == 100 ?

Reply:
it is not, the exception gets thrown only if it is < 0 and > 100

Reply:
aah right  I misread that